src/azure-account.ts

```diff
@@ -773,17 +785,47 @@ async function getEnvironments(): Promise<Environment[]> {
 
 	const config = workspace.getConfiguration('azure');
 	const ppe = config.get<Environment>('ppe');
-	if (ppe) {
-		return [
-			...staticEnvironments,
-			{
-				...ppe,
-				name: azurePPE
-			}
-		]
-	} else {
-		return staticEnvironments;
+	const apiProfile = config.get<Boolean>('target_azurestack_api_profile');
+	try {
+		if (ppe) {
+			const activeDirectoryUrl = ppe.activeDirectoryEndpointUrl.endsWith('/') ? ppe.activeDirectoryEndpointUrl.slice(0,-1) : ppe.activeDirectoryEndpointUrl;
+			const validateAuthority = activeDirectoryUrl.endsWith('/adfs') ? false : true;
+			if (apiProfile) {
+				let resourceMangerUrl = ppe.resourceManagerEndpointUrl;
+				resourceMangerUrl = resourceMangerUrl.endsWith('/') ? resourceMangerUrl.slice(0,-1) : resourceMangerUrl;
```

Comment:
To make it a little bit clearer why we're doing this, I think it would be good to extract everything inside this `if` block to a new function called something like `getAzureStackEnvironments`